enhancements/manifestlist/manifestlist-support.md

```diff
@@ -0,0 +1,404 @@
+images that support different architectures as well. By adding support for manifest lists we are
+also adding support for hosting these multi-architecture images in OpenShift's internal registry.
+
+### Goals
+
+- Allow users to import manifest lists using the `oc` client.
+- Allow users to execute pull-through of manifest lists.
+- Make image pruner aware of manifest lists.
+- Allow users to push manifest lists into the integrated registry.
+- Improve the `oc describe` command to present manifest list info.
+
+### Non-Goals
+
+- Modify OpenShift's web console to support manifest lists.
+- Designs OpenShift multi-architecture builds.
+- Provide facilities to build manifest lists.
+
+## Proposal
+
+The proposed approach involves allowing one Image struct to refer to multiple other images by
+their hashes. In such a scenario, an Image `A` would represent the manifest list containing among
+its properties a slice of other Images `B` and `C`, each of them being a Manifest for an image of
+a different platform.
+
+In other words, when seeing from an ImageStream point of view we would have a mapping that looks
+like this:
+
+```text
+ImageStream X -> Image A -> Image B
+                            Image C
+                            Image D
+
+```
+
+Here `Image A` would contain references to `B`, `C`, and `D`. As a manifest list is simply a
+reference to other manifests they do not have any extra information such as "environment
+variables", "entry point" or "open ports" among its fields, this extra information, therefore,
+still belongs to the "child images" (`B` and `C` for instance). Image A would serve as a means
+to group other Images.
+
+### User Stories
+
+#### Support for manifest list import
+
+As an OpenShift user
+I want to be able to import manifest lists
+So that I can refer to images supporting multiple platforms
+
+#### Support for manifest list pull-through
+
+As an OpenShift user
+I want to be able to pull through manifest lists
+So that I can cache multi-architecture images within my cluster
+
+#### Support for pushing manifest lists to OpenShift's image registry
+
+As an OpenShift user
+I want to be able to push manifest lists to Openshift's internal registry
+So that I can publish my multi-architecture images
+
+#### Support for manifest lists during prune
+
+As an OpenShift user
+I want to be able to automatically clean unused manifest lists
+So that I can keep my storage resource usage under control
+
+### Implementation Details/Notes/Constraints
+
+#### On pushing manifests into integrated registry
+
+Openshifft's internal registry will need to accept image pushes by SHA, not only by tag. The
+difference here is that the registry is not supposed to create an Image object every time it
+sees a push by SHA but only when it sees a push by tag. In the latter scenario the registry may
+need to create more than one image (if the pushed manifest is of type manifest list).
```

Comment:
Risk: the pruner won't see these images, the hard pruner doesn't know about its age and whether this manifest has been just uploaded for a manifest list.